static/js/NavSidebar.jsx

```diff
@@ -85,6 +87,59 @@ const TitledText = ({enTitle, heTitle, enText, heText}) => {
   </Module>
 };
 
+const RecentlyViewed = ({toggleSignUpModal, mobile}) => {
+   const [recentlyViewedItems, setRecentlyViewedItems] = useState(null);
+   const handleAllHistory = (e) => {
+    if (!Sefaria._uid) {
+      e.preventDefault();
+      toggleSignUpModal(SignUpModalKind.ViewHistory);
```

Comment:
I think perhaps this needs to be handled outside the component, i'm not sure I love that this is being passed into it.

Reply:
I don't see how to handle it outside the component.  Within the component is a link to the history page and if one is logged out when they click to go to their history page, we want to prompt them to sign up/sign in.  I can't think of another approach.  What do you think?